Library/Homebrew/extend/os/mac/keg_relocate.rb

```diff
@@ -111,6 +123,12 @@ def each_install_name_for(file, &block)
     dylibs.each(&block)
   end
 
+  def each_rpath_for(file, &block)
+    rpaths = file.rpaths
+    rpaths.reject! { |fn| fn =~ /^@(loader|executable)_path/ }
```

Comment:
I don't understand the need for this. All we do is `start_with?` checks anyway?

Reply:
This is a good question. I just copied the logic from here:

https://github.com/Homebrew/brew/blob/3e327626079e22f4becebb26b632ad431400105e/Library/Homebrew/extend/os/mac/keg_relocate.rb#L107-L112

I guess the idea was to save some `start_with?` checks, but that's just replacing one operation with another... I can remove it if you prefer. Though we should probably do the same with `each_install_name_for` unless there's a reason those checks belong there but not here.

Reply:
Actually, doing `reject!` is probably way less efficient than skipping straight to the `start_with?` checks.